lib/Qandidate/Toggle/Serializer.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the qandidate/toggle package.
+ *
+ * (c) Qandidate.com <opensource@qandidate.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Qandidate\Toggle;
+
+/**
+ * Representation of a serializer
+ *
+ * This class provides common ground for concrete serializer implementations
+ *
+ * @TODO: serialize() and deserialize() should be part of abstract interface
+ */
+abstract class Serializer
```

Comment:
Yep. This one got into the PR since I wasn't using a branch for the PR itself. `AssertionHelper` then. 

When we're talking about signatures - these would have to be unified, and argument validation should be carried out by child classes. I don't think it's the time for this change yet - unless I'll need to do some generic mocks.